code/modules/research/reward_system/_reward_system.dm

```diff
@@ -0,0 +1,106 @@
+// Filled out in code/modules/research/techweb/_techweb_node.dm Initialize()
+// Lists of IDs (strings) which can be accessed via SSresearch.techweb_node_by_id(id)
+// Assoc list, id = TRUE
+GLOBAL_LIST_EMPTY(security_nodes)
+GLOBAL_LIST_EMPTY(medical_nodes)
+GLOBAL_LIST_EMPTY(cargo_nodes)
+GLOBAL_LIST_EMPTY(service_nodes)
+GLOBAL_LIST_EMPTY(science_nodes)
+GLOBAL_LIST_EMPTY(engineering_nodes)
+// A reward system framework for crew: By doing their jobs, they get to progress towards research nodes for their department
+// Must be implemented very carefully to avoid exploitation, it should be very boring or annoying to try and exploit it at least
+/obj/machinery/computer/department_reward
+	name = "departmental research console (ERROR)"
+	desc = "A primitive version of the famous R&D console used to unlock department-relevant research."
+	icon_screen = "rdcomp"
+	icon_keyboard = "rd_key"
+	circuit = /obj/item/circuitboard/computer/department_reward
+	light_color = LIGHT_COLOR_BLUE
+	var/datum/techweb/linked_techweb
+	var/list/nodes_available = null // pass by reference, not .Copy()
+	var/points = 0
+	var/department_display = "ERROR"
+
+/obj/machinery/computer/department_reward/Initialize(mapload)
+	. = ..()
+	linked_techweb = SSresearch.science_tech
+
+/obj/machinery/computer/department_reward/proc/check_reward(delta_time)
+	return 0
+
+/obj/machinery/computer/department_reward/process(delta_time)
+	if(stat & (NOPOWER|BROKEN))
+		return // we dont return PROCESS_KILL because it's already handled in /obj/machinery
```

Comment:
comment doesn't really make sense

you probably don't want to ever use PROCESS_KILL